### PR TITLE
[TACHYON-879] Adding unit tests for tachyon.util.FormatUtils

### PR DESCRIPTION
**Pull request by jsimsa**
This PR adds unit tests for tachyon.util.FormatUtils and implements related minor fixes and refactors (for ease of testing).

### CODE REVIEW

File: common/src/main/java/tachyon/util/FormatUtils.java [hunk 1]
```diff
@@ -31,31 +31,33 @@
...
 
-  public static void printTimeTakenMs(long startTimeMs, Logger logger, String message) {
-    logger.info(message + " took " + (CommonUtils.getCurrentMs() - startTimeMs) + " ms.");
+  public static String formatTimeTakenMs(long startTimeMs, String message) {
+    return message + " took " + (CommonUtils.getCurrentMs() - startTimeMs) + " ms.";
   }
 
-  public static void printTimeTakenNs(long startTimeNs, Logger logger, String message) {
-    logger.info(message + " took " + (System.nanoTime() - startTimeNs) + " ns.");
+  public static String formatTimeTakenNs(long startTimeNs, String message) {
```
Comment (apc999) [inline common/src/main/java/tachyon/util/FormatUtils.java]: Can you javadoc this method and the above one?

File: common/src/test/java/tachyon/util/FormatUtilsTest.java [hunk 2]
```diff
@@ -22,6 +30,144 @@
 
 public class FormatUtilsTest {
   @Test
+  public void parametersToStringTest() {
+    class TestCase {
```
Comment (apc999) [inline common/src/test/java/tachyon/util/FormatUtilsTest.java]: shall we make class `TestCase` private?

File: common/src/main/java/tachyon/util/FormatUtils.java [hunk 3]
```diff
@@ -31,31 +31,33 @@
...
   public static String parametersToString(Object... objs) {
     StringBuilder sb = new StringBuilder("(");
-    for (int k = 0; k < objs.length; k ++) {
-      if (k != 0) {
+    for (Object obj : objs) {
+      if (sb.length() != 1) {
         sb.append(", ");
       }
-      sb.append(objs[k].toString());
+      sb.append(obj.toString());
```
Comment (apc999) [inline common/src/main/java/tachyon/util/FormatUtils.java]: shall we check if `obj` is null?

**Comment by apc999** [comment]
Thanks for the patch to improve test health, @jsimsa .
I left some minor comments. The PR in general looks good.

File: common/src/main/java/tachyon/util/FormatUtils.java (hunk 1 above)
Comment (jsimsa) [inline common/src/main/java/tachyon/util/FormatUtils.java]: Done.

File: common/src/test/java/tachyon/util/FormatUtilsTest.java (hunk 2 above)
Comment (jsimsa) [inline common/src/test/java/tachyon/util/FormatUtilsTest.java]: TestCase is a local class and as such cannot be declared private (http://docstore.mik.ua/orelly/java-ent/jnut/ch03_11.htm). Are you suggesting I reorganize the code to make it private? I think the current organization makes sense as each TestCase class only pertains to the method in which it is defined.

File: common/src/main/java/tachyon/util/FormatUtils.java (hunk 3 above)
Comment (jsimsa) [inline common/src/main/java/tachyon/util/FormatUtils.java]: Done and added a test for that.

**Comment by jsimsa** [comment]
Thanks Bin. Addressed your comments. PTAL

File: common/src/main/java/tachyon/util/FormatUtils.java [hunk 4]
```diff
@@ -29,33 +29,68 @@
...
+   * @return formatted string with the elapsed time
+   */
+  public static String formatTimeTakenMs(long startTimeMs, String message) {
+    return message + " took " + (CommonUtils.getCurrentMs() - startTimeMs) + " ms.";
   }
 
-  public static void printTimeTakenNs(long startTimeNs, Logger logger, String message) {
-    logger.info(message + " took " + (System.nanoTime() - startTimeNs) + " ns.");
+  /**
+   * Format time elapsed since the given start time.
```
Comment (gpang) [inline common/src/main/java/tachyon/util/FormatUtils.java]: Comment that this is in ns.

File: common/src/main/java/tachyon/util/FormatUtils.java [hunk 5]
```diff
@@ -29,33 +29,68 @@
...
     }
-
-    LOG.info(sb.toString());
+    return sb.toString();
   }
 
-  public static void printTimeTakenMs(long startTimeMs, Logger logger, String message) {
-    logger.info(message + " took " + (CommonUtils.getCurrentMs() - startTimeMs) + " ms.");
+  /**
+   * Format time elapsed since the given start time.
```
Comment (gpang) [inline common/src/main/java/tachyon/util/FormatUtils.java]: Mention this is in ms

File: common/src/main/java/tachyon/util/FormatUtils.java [hunk 6]
```diff
@@ -29,33 +29,68 @@
...
+   *         of the individual objects
+   */
   public static String parametersToString(Object... objs) {
     StringBuilder sb = new StringBuilder("(");
-    for (int k = 0; k < objs.length; k ++) {
-      if (k != 0) {
-        sb.append(", ");
+    if (objs != null) {
+      for (Object obj : objs) {
+        if (sb.length() != 1) {
```
Comment (gpang) [inline common/src/main/java/tachyon/util/FormatUtils.java]: Can obj.toString() ever return an empty string? Actually, probably, if the actual emtpy string is a parameter.

If so, then this condition would behave differently from the previous code. What should happen if toString() returns an empty string? There should still be an entry for it?

And, can you add a test case for this case?

**Comment by gpang** [comment]
Thanks for the PR! I left a few comments.

File: common/src/main/java/tachyon/util/FormatUtils.java (hunk 5 above)
Comment (jsimsa) [inline common/src/main/java/tachyon/util/FormatUtils.java]: Done.

File: common/src/main/java/tachyon/util/FormatUtils.java (hunk 4 above)
Comment (jsimsa) [inline common/src/main/java/tachyon/util/FormatUtils.java]: Done.

File: common/src/main/java/tachyon/util/FormatUtils.java (hunk 6 above)
Comment (jsimsa) [inline common/src/main/java/tachyon/util/FormatUtils.java]: Done. Let me know if you'd prefer to show a something else than an empty space for empty string. As it is, an empty array and an array containing an empty string produce the same output: ().

**Comment by jsimsa** [comment]
Thanks @gpang. Addressed your comments. PTAL

File: common/src/test/java/tachyon/util/FormatUtilsTest.java [hunk 7]
```diff
@@ -22,6 +30,147 @@
...
+      public TestCase(String expected, Object[] objs) {
+        mExpected = expected;
+        mInput = objs;
+      }
+    }
+
+    List<TestCase> testCases = new LinkedList<TestCase>();
+    testCases.add(new TestCase("()", null));
+    testCases.add(new TestCase("(null)", new Object[] {null}));
+    testCases.add(new TestCase("()", new Object[] {""}));
```
Comment (gpang) [inline common/src/test/java/tachyon/util/FormatUtilsTest.java]: Could you add one more test case with multiple empty strings in the beginning of the array? Like {"", "", ""}?

**Comment by gpang** [comment]
Just a minor comment about a test case, but looks good!

File: common/src/test/java/tachyon/util/FormatUtilsTest.java (hunk 7 above)
Comment (jsimsa) [inline common/src/test/java/tachyon/util/FormatUtilsTest.java]: Done.

**Comment by gpang** [comment]
LGTM
